src/main/java/org/opensearch/knn/quantization/factory/QuantizerRegistry.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.factory;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.opensearch.knn.quantization.models.quantizationParams.QuantizationParams;
+import org.opensearch.knn.quantization.quantizer.Quantizer;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * The QuantizerRegistry class is responsible for managing the registration and retrieval
+ * of quantizer instances. Quantizers are registered with specific quantization parameters
+ * and type identifiers, allowing for efficient lookup and instantiation.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+final class QuantizerRegistry {
+    // ConcurrentHashMap for thread-safe access
+    private static final Map<String, Quantizer<?, ?>> registry = new ConcurrentHashMap<>();
+
+    /**
+     * Registers a quantizer with the registry.
+     *
+     * @param paramIdentifier the unique identifier for the quantization parameters
+     * @param quantizer       an instance of the quantizer
+     */
+    static void register(final String paramIdentifier, final Quantizer<?, ?> quantizer) {
+        // Ensure that the quantizer for this identifier is registered only once
+        registry.putIfAbsent(paramIdentifier, quantizer);
+    }
```

Comment:
should we throw an exception if using a same paramIdentifier a quantizer is already registered?